t/form_with_fields.t

```diff
@@ -36,6 +36,67 @@ ok( $mech->success, "Fetched $uri" ) or die q{Can't get test page};
 }
 
 {
+    my $w; local $mech->{onwarn} = sub { warn $w if defined $w; $w = $_[0] };
+    my $form = $mech->form_with_fields('4a', '4b');
+
+    isa_ok( $form, 'HTML::Form' );
+    is($form->attr('name'), '4th_form_1', 'fourth form matches');
+    is($w, 'There are 2 forms with the named fields.  The first one was used.', 'warning on ambiguous match');
```

Comment:
It might be worth using something like https://metacpan.org/pod/Test::Warnings to test the warnings here.